lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -64,31 +65,128 @@ module Billing #:nodoc:
     #     :tax2 => 100,
     #     :custom => 'reference one'
     #   )
+    #
+    # Modification on 2018-03-12:
+    # New feature: integrated the ipp gateway into beanstream gateway, use this class as an unified entrance for both the gateways
+    # Get the :region field from options parameter, :region: 0 - US & CA, 1 - AU
+    # Proess payment request from US & CA using beanstream service (default)
+    # Proess payment request from AU using ipp service
+    # Will remove ipp gateway file
+    #
+
     class BeanstreamGateway < Gateway
-      include BeanstreamCore
+      attr_accessor :region
+      include Empty
+
+      # Only <tt>:login</tt> is required by default,
+      # which is the merchant's merchant ID. If you'd like to perform void,
+      # capture or refund transactions then you'll also need to add a username
+      # and password to your account under administration -> account settings ->
+      # order settings -> Use username/password validation
+      def initialize(options = {})
+        #region: 0 - US & CA, 1 - AU & NZ
+        @region = empty?(options[:region]) ? 0 : options[:region]
+
+        # I want to add this logic to judge the region from shipping & billing address
+        # but don't know if it is a correct way.
+        # I think this can be decided by shopify which will integrate with.
```

Comment:
@lumiit-adam if I understand correctly, `options[:region]` is a new option what would need to be passed in, and wasn't used previouly?

Reply:
If there is other way can pass the parameter without add new option, please tell me.